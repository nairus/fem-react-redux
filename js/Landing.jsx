// @flow

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Search from './Search';

class Landing extends Component {
  state = {
    searchTerm: ''
  };
  handleKeyPress = (event: SyntheticKeyboardEvent & { target: HTMLInputElement }) => {
    // Change the state if enter key is pressed.
    if (event.charCode === 13) {
      this.setState({ searchTerm: event.target.value });
    }
  };
  props: {
    shows: Array<Show>
  };
  render() {
    if (this.state.searchTerm) {
      return <Search showSearch search={this.state.searchTerm} shows={this.props.shows} />;
    }

    // Otherwise, render the homepage
    return (
      <div className="landing">
        <h1>svideo</h1>
        <input type="text" placeholder="Search" onKeyPress={this.handleKeyPress} />
        <Link to="/search">or Browse All</Link>
      </div>
    );
  }
}

const mapStateToProps = state => ({ searchTerm: state.searchTerm });

export default connect(mapStateToProps)(Landing);
