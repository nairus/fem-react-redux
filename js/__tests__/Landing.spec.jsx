import React from 'react';
import { shallow } from 'enzyme';
import { Unwrapped as UnwrappedLanding } from '../Landing';

test('Landing render correctly', () => {
  const component = shallow(<UnwrappedLanding />);
  expect(component).toMatchSnapshot();
});
