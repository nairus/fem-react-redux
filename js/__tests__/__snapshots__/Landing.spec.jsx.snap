// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing render correctly 1`] = `
<div
  className="landing"
>
  <h1>
    svideo
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      placeholder="Search"
      type="text"
    />
  </form>
  <Link
    replace={false}
    to="/search"
  >
    or Browse All
  </Link>
</div>
`;
